packages/build/src/plugins/mutations.js

```diff
@@ -1,23 +1,18 @@
 'use strict'
 
-const { set } = require('dot-prop')
-
 const { addErrorInfo } = require('../error/info')
+const { setProp } = require('../utils/set')
 
 const { getPropName } = require('./config_prop_name')
 const { EVENTS } = require('./events')
 
 // Apply a series of mutations to `netlifyConfig`.
-// Also normalize it.
+// Also denormalize it.
 const applyMutations = function (priorityConfig, configMutations) {
-  configMutations.forEach(({ keys, value, event }) => {
-    applyMutation({ priorityConfig, keys, value, event })
-  })
-  return priorityConfig
+  return configMutations.reduce(applyMutation, priorityConfig)
```

Comment:
Use shallow copy instead of direct mutations on `configMutations` to avoid unexpected mutations from a caller standpoint.